plugin/src/main/java/io/fabric8/maven/plugin/SetupMojo.java

```diff
@@ -0,0 +1,295 @@
+                }
+            }
+            getLog().info("Updating the pom " + pom);
+            try {
+                DomHelper.save(doc, pom);
+            } catch (Exception e) {
+                getLog().error("Failed to update pom " + pom + ". " + e, e);
+                throw new MojoExecutionException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private boolean createOrUpdateFabric8MavenPlugin(Document doc) throws MojoExecutionException {
+        boolean updated = false;
+        String latestVersion;
+        try {
+            latestVersion = MavenUtil.getVersion(PLUGIN_GROUPID, PLUGIN_ARTIFACTID);
+        } catch (IOException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+        // lets check that there's a property defined
+        String currentVersion = project.getProperties().getProperty(FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
+        String versionExpression = "${" + FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY + "}";
+        Element configuration = null;
+        Element fmpPlugin = findPlugin(doc, PLUGIN_GROUPID, PLUGIN_ARTIFACTID);
+        boolean pluginInserted = fmpPlugin == null;
+        if (updateVersion || pluginInserted) {
+            if (useVersionProperty) {
+                Element documentElement = doc.getDocumentElement();
+                Element properties = firstChild(documentElement, "properties");
+                if (properties == null) {
+                    properties = addChildAfter(appendAfterLastElement(properties, doc.createTextNode("\n      ")), "properties");
+                }
+                if (Strings.isNullOrBlank(currentVersion)) {
+                    addChildAfter(appendAfterLastElement(properties, doc.createTextNode("\n    ")), FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY, latestVersion);
+                    updated = true;
+                } else if (!Objects.equals(currentVersion, latestVersion)) {
+                    Element propertyElement = DomHelper.firstChild(properties, FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
+                    if (propertyElement != null) {
+                        propertyElement.setTextContent(latestVersion);
+                        updated = true;
+                    }
+                }
+            } else {
+                versionExpression = latestVersion;
+            }
+        }
+
+        if (fmpPlugin == null) {
+            fmpPlugin = findOrAddPlugin(doc, PLUGIN_GROUPID, PLUGIN_ARTIFACTID, versionExpression, configuration);
+            updated = true;
+        } else {
+            if (updateVersion || pluginInserted) {
+                String version = DomHelper.firstChildTextContent(fmpPlugin, "version");
+                if (version == null || !version.equals(versionExpression)) {
+                    Element versionElement = DomHelper.firstChild(fmpPlugin, "version");
+                    if (versionElement == null) {
```

Comment:
Still not sure whether we unconditionally should add a version in this case, since the version could also be specified in a `<pluginManagement>` section. Adding the version here would override the version specified in the pluginManagement section and might lead to unexpected behaviour.

It would be much better to update the `<pluginManagement>` 's version if existing.

And if we want to be really clever we should check also for parent pom's and boms (for both, a direct plugin declaration and the `<pluginManagement>`)